shared/state.rs

```diff
@@ -301,4 +304,31 @@ impl State {
 
         Ok(())
     }
+
+    /// Checkek that any Queued tasks dont have any failed dependencies, otherwise marked it as failed
+    pub fn update_dependencies(&mut self) {
+        let has_failed_deps: Vec<_> = self
+            .tasks
+            .iter()
+            .filter(|(_, task)| task.status == TaskStatus::Queued)
+            .filter_map(|(id, task)| {
```

Comment:
As far as I can see, you only look for single level dependencies.

Scenario:
```
a -> b -> c
```
`a` fails, `b` detects `a` as dependency and fails.
`c` however didn't detect `b` as a failed dependency before.
`c` is started.

As I said, we will probably need proper dependency graph handling or you have to implement a simple recursive lookup for now.

Reply:
This will work : 
`a` fails, `b` detects `a` as dependency and fails.
`c` however didn't detect `b` as a failed dependency before.
`c` is **not** started, because `b` is not in `Done` state.
`c` detects that `b` failed and is now failed as well.

It takes 1 more internal tick, but it won't start task by mistake.

Reply:
Yess, you're right. Sorry, forgot that tasks won't start until their dependencies are finished